src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -2649,6 +2649,13 @@ private bool CheckInArgTypes(TexlNode left, TexlNode right)
                             return false;
                         }
 
+                        // restrict scalar in multiselect optionset table. (Not TableA.OptionsetColumn)
+                        if (typeRight.IsMultiSelectOptionSet() && typeLeft.AssociatedDataSources?.Count == 0)
+                        {
+                            _txb.ErrorContainer.EnsureError(DocumentErrorSeverity.Severe, left, TexlStrings.ErrBadType_Type, typeLeft.GetKindString());
```

Comment:
This seems arbitrary. Why do we care whether the lhs is from a data source or not? I understand for delegation, but what about in memory eval?